File: javascript/easy/selfDividingNumbers.js
/**
 * @param {number} left
 * @param {number} right
 * @return {number[]}
 */
var selfDividingNumbers = function(left, right) {
    const res = new Array();
    while(left <= right) {
        if(isSelfDividing(left)) {
            res.push(left);
        }
        left++;
    }
    return res;
};

var isSelfDividing = function(num) {
    let x = num;
    while(x > 0) {
        let digit = (x % 10);
        if(digit == 0 || (num % digit) != 0) {
            return false;
        }
        x = parseInt(x/10);
    }
    return true;
};

File: ruby/easy/self_dividing_numbers.rb
# @param {Integer} left
# @param {Integer} right
# @return {Integer[]}
def self_dividing_numbers(left, right)
    res = []
    while(left <= right)
        res.push(left) if is_self_dividing(left)
        left += 1
    end
    return res
end

def is_self_dividing(num)
    x = num
    while x > 0
        d = (x % 10)
        return false if (d == 0) || ((num % d) != 0)
        x /= 10
    end
    return true
end
